backend/app/views/spree/admin/products/_form.html.erb

```diff
@@ -32,7 +32,8 @@
       <div data-hook="admin_product_form_price">
         <%= f.field_container :price do %>
           <%= f.label :price, class: 'required' %>
-          <%= render "spree/admin/shared/number_with_currency", f: f, amount_attr: :price, currency: @product.find_or_build_master.default_price.currency %>
+          <%= render "spree/admin/shared/number_with_currency", f: f, amount_attr: :price,
+                     currency: (@product.find_or_build_master.default_price || Spree::Config.default_pricing_options).currency %>
```

Comment:
Would this line not always just return `Spree::Config.currency`? IE: We could probably just save all the initialization and just use the default currency here and call it a day...